src/components/App.js

```diff
@@ -2,13 +2,42 @@ import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
 /* eslint-disable no-unused-vars */
-import calculate from '../logic/calculate';
-
-const App = () => (
-  <div id="main-div">
-    <Display />
-    <ButtonPanel />
-  </div>
-);
+import main from '../logic/calculate';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      next: null,
+      operation: null,
+      total: null,
+    };
+  }
+
+  handle(btname) {
+    const { next, operation, total } = main(this.state, btname);
+    this.setState({
+      operation,
+      total,
+      next,
+    });
+  }
+
+  render() {
+    const { next, operation, total } = this.state;
+    let value = null;
+    if (operation === null) {
+      value = total || '0';
+    } else {
+      value = next || total;
+    }
+    return (
+      <div id="main-div">
+        <Display value={value.toString()} />
+        <ButtonPanel onClick={btname => this.handle(btname)} />
```

Comment:
there should be a prop that handles the clickhandler, you are calling onclick directly that is not the desired behaviour ![screen](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone5/app/passHandleClick.png)